File: 0242-valid-anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        f={}
        for ch in s:
            f[ch]=f.get(ch,0)+1
        for ch in t:
            if ch not in s:
                return False
            f[ch]-=1
        for v in f.values():
            if v!=0:
                return False
        return True        

File: 0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character,Integer> m=new HashMap<>();
        if (s.length()!=t.length()){
            return false;
        }
        for(int i=0;i<s.length();i++){
            char ch=s.charAt(i);
            m.put(ch,m.getOrDefault(ch,0)+1);
        }
        for(int i=0;i<t.length();i++){
            char ch=t.charAt(i);
            if(!m.containsKey(ch) || m.get(ch)==0){
                return false;
            }
            m.put(ch,m.get(ch)-1);
        }
        return true;
    }
}
